### ISSUE TITLE
Regarding to your joint goal accuracy

### ISSUE THREAD

**Issue by jasonwu0731** [SKTBrain/SUMBT#7]
Hello,

I found this part a bit weird (line 989) in the evaluation. 

https://github.com/SKTBrain/SUMBT/blob/master/code/main-multislot.py#L989

When you compute joint goal accuracy, the score at each turn can only be 0 or 1. But here since that you are using "/" instead of "//" to get a floating number? Can you double check if I am wrong? Many thanks.

**Comment by jasonwu0731** [comment]
Also, when we are going to check the accuracy, I think we need to change back to "string" and then do the evaluation (no matter direct match or fuzzy match). Because right now since the dataset has many labels in the test set are not "NONE" but you are assuming they are "NONE"?

**Comment by hwaranlee** [comment]
Hi, Thanks for reviewing our code.

For the first issue, since the 'accuracy' is a torch.BoolTensor, 
https://github.com/SKTBrain/SUMBT/blob/011127be0587f942e95b6dd8e6e15195577a637c/code/main-multislot.py#L985
(torch.sum(accuracy, 1) / slot_dim) is a torch.LongTensor,
https://github.com/SKTBrain/SUMBT/blob/011127be0587f942e95b6dd8e6e15195577a637c/code/main-multislot.py#L989
which means that the joint accuracy is 1 when all slots are correct.

For the second issue, please refer to the script: 
SUMBT/data/multiwoz/original/convert_to_glue_format.py
(you should unzip the original.zip file)
Several slot values are matched to the values in the ontology by the rules. 
In this work, our model does _not generate_ slot values but _predict (or select)_ among the slot value candidates. I agree that there is a noisy annotation issue in the dataset, however, we assumed the dataset is clean enough and followed the convention.

**Comment by jasonwu0731** [comment]
hi @hwaranlee ,

Thanks for your quick reply. Now I got it. It is a long tensor so divide by slot_dim will be either 1 or 0. For the second one, I'd prefer to create a new ontology based on the state labels since the original provided ontology is not completed.

Anyway, I got your point and thanks again for your prompt reply.

**Comment by bumbumya** [comment]
Hello,@hwaranlee  @jasonwu0731 
I found the joint accuracy is not 0 when some slots are correct and others are wrong,  it's a floating number, did I get something wrong?
